src/gesture-handler/GestureDetector.tsx

```diff
@@ -0,0 +1,337 @@
+import React, { useEffect } from 'react';
+import { Dimensions, findNodeHandle } from 'react-native';
+import {
+  GestureDetector,
+  Gesture,
+  PanGestureHandlerEventPayload,
+  GestureUpdateEvent,
+} from 'react-native-gesture-handler';
+import {
+  useSharedValue,
+  measure,
+  startScreenTransition,
+  finishScreenTransition,
+  ScreenTransition,
+  makeMutable,
+} from 'react-native-reanimated';
+import type { GestureProviderProps } from 'src/native-stack/types';
+import { getShadowNodeWrapperAndTagFromRef, isFabric } from './fabricUtils';
+
+type RNScreensTurboModuleType = {
+  startTransition: (stackTag: number) => {
+    topScreenTag: number;
+    belowTopScreenTag: number;
+    canStartTransition: boolean;
+  };
+  updateTransition: (stackTag: number, progress: number) => void;
+  finishTransition: (stackTag: number, isCanceled: boolean) => void;
+};
```

Comment:
Why do you type it this way, instead of declaring these methods on the module spec itself in `src/fabric/NativeScreensModule.ts`?

^ treat this as a question for a meeting ;D: why do we not expose these module's methods in the spec, as you need to consume them in reanimated any way? 

My current understanding is that you do not expose anything in public API (by not declaring any methods in the TS module spec anyway) & you just access the module's methods that were injected into the runtime's global object during module initialisation.

Why is this preferred instead of just defining methods on module, implementing them & using them from JS through module itself, not global object?